### DISCUSSION TITLE
How do I crop or slice a specific part of image?

### DISCUSSION THREAD

**Discussion by smartdev007** [ImageMagick/ImageMagick#2134] [Help] [answered]
![DBTC5MaW0AAu3C8](https://user-images.githubusercontent.com/16380239/84458076-31ad3a80-ac32-11ea-99ed-fd3c8cbcf87a.jpg)

For example if I want to crop the first part of the image then how would I do so? I tried using crop command but couldn't figure out the dimensions to pass.

**Comment by fmw42** [comment, marked as answer]
-crop WxH+X+Y  where W is the width of the cropped region, H is the height of the cropped region, X and Y represent the top-left corner of the cropped region relative to the top-left corner of the full image.  So if I want a 100x100 size region that starts at 10,10 from the top left corner of the original image, the crop would be -crop 100x100+10+10.
 
`convert stockholm.jpg -crop 100x100+10+10 +repage stockholm_crop.jpg`

![stockholm_crop](https://user-images.githubusercontent.com/13458218/84458826-b9874a80-ac1a-11ea-8911-442885a1d48d.jpg)

If on IM 7, use magick in place of convert.

**Reply by smartdev007** [reply]
How do I determine the size of the cropped region though? I mean how would I know the size of the 1st rank row in the leaderboard of the image?

**Reply by fmw42** [reply]
You need to use a GUI based tool such as GIMP or Photoshop or some other tool that shows the coordinates of the cursor. If on Linux, when you display the image in ImageMagick (display some_image_name), you can click on the image or hold the mouse button down and it will show the coordinates. Otherwise, ImageMagick is just a command line tool.  I am on a Mac and use GraphicConvert to find coordinates. Then I supply them to my ImageMagick commands. On Windows, there are non-Imagemagick viewing tools such as Irfanview that will read the cursor locations.

**Reply by smartdev007** [reply]
Thanks for the help so far. So I downloaded the Gimp and got x and y coordinate of the top left position of the area that I want. But then how do I set the limit on how far the tool should crop my image? Like how do I set the top right corner, down right corner and down left corner limits?

Edit: nvm, I figured it out. Gimp shows the size of the area that I want to crop as well so just passed the size and position as parameters and got the result. Thanks a lot for the help! :)

**Reply by fmw42** [reply]
You set the width and height and not the other corners. You can use GIMP to locate those corners and then do the calculations (subtractions) to convert to Width and Height.

**Reply by smartdev007** [reply]
Yeah, I got the outout, thanks a lot! Also, dpi of the output file comes up as 0. Any idea how I can increase the dpi to 300? I want to pass the result to tesseract ocr but it's not able to read the correct text.

**Reply by fmw42** [reply]
What was your exact command line?  What is your IM version and platform, since syntax may vary.

**Reply by smartdev007** [reply]
Command to crop image: `convert DBTC5MaW0AAu3C8.jpg -crop 762x41+32+100 test2.png`
Command to read the output file from tesseract: `tesseract test2.png stdout`
IM version:  6.9.7-4
Platform: Ubuntu 18.04

**Reply by fmw42** [reply]
`convert DBTC5MaW0AAu3C8.jpg -crop 762x41+32+100 -units pixelsperinch -density 300 test2.png`

But for raster images, the density will have no effect on the quality of the cropped image. Density for raster images is only used when printing to decide how large in physical units such as inches the image will be printed.

Density is important when reading vector files such as PDF and the larger the density the larger the vector image will be rasterized into pixels.

**Reply by fmw42** [reply]
One other thing. JPG is not a good format if you want to have best quality. It is a lossy compressed format. Every time you read and write the JPG, it loses quality, even when you just crop it. If you want to maintain best quality, use PNG or TIFF and add +repage after the crop to remove the virtual canvas that maintains the original size.

**Reply by smartdev007** [reply]
I converted file to .png and then tried out the reading from tesseract but still got the same result. What would you recommend so that I get the exact text that's in the image?

**Reply by fmw42** [reply]
Once you have JPG, you have lost quality. Better to create/scan/etc as PNG or TIFF. Please show the image you created to use for tesseract. You should likely convert it so that the text is black on a white background. What was the input image you started with? Can you post that?

**Comment by smartdev007** [comment]
![input](https://user-images.githubusercontent.com/16380239/84533695-ad50cb00-acb6-11ea-9b1d-a5d405cd51ea.png)

![test2](https://user-images.githubusercontent.com/16380239/84533696-ade96180-acb6-11ea-9af4-9d26deece8ab.png)
This is my input image and cropped image

**Reply by fmw42** [reply]
Try this result with tesseract. Adjust the threshold percent as needed.

`convert crop.png -auto-level -negate -threshold 65% crop_processed.png`

![crop_processed](https://user-images.githubusercontent.com/13458218/84535209-31e41f00-aca0-11ea-93c7-f1c0eed39fb8.png)

**Reply by smartdev007** [reply]
It worked. Thanks a lot for the help and guiding! :)

**Reply by fmw42** [reply]
Tesseract generally works best with black lettering on white background.

**Reply by smartdev007** [reply]
Yeah, results are better with black letters and white background. Thanks for the tip!

**Reply by smartdev007** [reply]
> 
> 
> Try this result with tesseract. Adjust the threshold percent as needed.
> 
> `convert crop.png -auto-level -negate -threshold 65% crop_processed.png`
> 
> ![crop_processed](https://user-images.githubusercontent.com/13458218/84535209-31e41f00-aca0-11ea-93c7-f1c0eed39fb8.png)

Is there any way of doing this in python?

**Reply by fmw42** [reply]
Yes. Ask on Stackoverflow in the Python Wand (which uses ImageMagick) or Python OpenCV forums.
